platforms/documentation/docs/src/docs/userguide/reference/gradle_wrapper.adoc

```diff
@@ -308,6 +308,27 @@ distributionUrl=https://username:password@somehost/path/to/gradle-distribution.z
 This can be used in conjunction with a proxy, authenticated or not.
 See <<networking.adoc#sec:accessing_the_web_via_a_proxy,Accessing the web via a proxy>> for more information on how to configure the `Wrapper` to use a proxy.
 
+WARNING: If you use systemProp.gradle.wrapperUser and systemProp.gradle.wrapperPassword to download the wrapper from a private server, your credentials can be leaked if you use build that will download the wrapper from another server.
+To prevent this, you can take the hostname of the private server, replace all dots with underscores, and use it as a prefix for the system properties.
+
+Example: If the private server hostname is `your.private-server.com`, you can use the following system properties:
+
+[source,properties]
+----
+systemProp.gradle.your_private-server_com.wrapperUser=username
+systemProp.gradle.your_private-server_com.wrapperPassword=password
+----
+
+To use a static HTTP Bearer Token, add the following line to the system properties file:
+
+[source,properties]
+----
+systemProp.gradle.your_private-server_com.wrapperToken=some-api-token
+----
+
+If a wrapperToken is specified, it will be used instead of the username and password.
+It is recommended to specify the hostname in the property to prevent the token from being accidentally sent to a foreign server
```

Comment:
```suggestion
It is strongly recommended to include the hostname in the property to prevent the token from being accidentally sent to an unintended server.
```